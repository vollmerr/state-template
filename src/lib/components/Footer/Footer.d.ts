import * as React from 'react';

import * as types from '../../utils/types';

export interface FooterProps {
  /** Link to use for 'contact us' link */
  contactLink: types.ContactLink;
}

declare class Footer extends React.Component<FooterProps, {}> {}

export default Footer;
