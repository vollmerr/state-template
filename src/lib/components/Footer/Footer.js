import React from 'react';

import * as propUtils from '../../utils/propTypes';
import A from '../A';

const externalLink = {
  target: '_blank',
  rel: 'noopener noreferrer',
};

class Footer extends React.PureComponent {
  getCopyRight = () => `Copyright \u00A9 ${new Date().getFullYear()}`

  render() {
    const { contactLink } = this.props;

    return (
      <>
        <footer className="global-footer">
          <div className="container">
            <div className="row">
              <div className="three-quarters">
                <ul className="footer-links">
                  <li><a href="http://www.ca.gov/Use" {...externalLink}>Conditions of Use</a></li>
                  <li><a href="http://www.ca.gov/Privacy" {...externalLink}>Privacy Policy</a></li>
                  <li><a href="http://www.ca.gov/Accessibility" {...externalLink}>Accessibility</a></li>
                  <li><A {...contactLink} /></li>
                </ul>
              </div>
              <div className="quarter text-right">
                <ul className="socialsharer-container">
                  <li>
                    <a href="https://www.flickr.com/groups/californiagovernment" {...externalLink}>
                      <span className="ca-gov-icon-flickr" aria-hidden="true" />
                      <span className="sr-only">Flickr</span>
                    </a>
                  </li>
                  <li>
                    <a href="https://www.pinterest.com/cagovernment/" {...externalLink}>
                      <span className="ca-gov-icon-pinterest" aria-hidden="true" />
                      <span className="sr-only">Pinterest</span>
                    </a>
                  </li>
                  <li>
                    <a href="https://twitter.com/cagovernment" {...externalLink}>
                      <span className="ca-gov-icon-twitter" aria-hidden="true" />
                      <span className="sr-only">Twitter</span>
                    </a>
                  </li>
                  <li>
                    <a href="https://www.youtube.com/user/californiagovernment" {...externalLink}>
                      <span className="ca-gov-icon-youtube" aria-hidden="true" />
                      <span className="sr-only">YouTube</span>
                    </a>
                  </li>
                </ul>
              </div>
            </div>
          </div>

          <div className="copyright">
            <div className="container">
              {this.getCopyRight()}
            </div>
          </div>
        </footer>

        <div className="decoration-last">&nbsp;</div>
      </>
    );
  }
}

Footer.propTypes = {
  /** Link to use for 'contact us' link */
  contactLink: propUtils.contactLink.isRequired,
};

export default Footer;
