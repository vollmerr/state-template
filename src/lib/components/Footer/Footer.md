```jsx
  <Footer contactLink={{ to: '/', text: 'Contact Link!' }}/>
```
