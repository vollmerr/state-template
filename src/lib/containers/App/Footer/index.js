import React from 'react';
import { Link } from 'react-router-dom';
import T from 'prop-types';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import * as routerSelectors from '../../Routing/selectors';

const externalLink = {
  target: '_blank',
  rel: 'noopener noreferrer',
};

class Footer extends React.PureComponent {
  getCopyRight = () => `Copyright \u00A9 ${new Date().getFullYear()}`

  render() {
    const { contactLink } = this.props;

    return (
      <React.Fragment>
        <footer id="footer" className="global-footer">
          <div className="container">
            <div className="row">
              <div className="three-quarters">
                <ul className="footer-links">
                  <li><a href="#skip-to-content">Back to Top</a></li>
                  <li><a href="http://www.ca.gov/Use" {...externalLink}>Conditions of Use</a></li>
                  <li><a href="http://www.ca.gov/Privacy" {...externalLink}>Privacy Policy</a></li>
                  <li><a href="http://www.ca.gov/Accessibility" {...externalLink}>Accessibility</a></li>
                  <li><Link to={contactLink}>Contact Us</Link></li>
                </ul>
              </div>
              <div className="quarter text-right">
                <ul className="socialsharer-container">
                  <li>
                    <a href="https://www.flickr.com/groups/californiagovernment">
                      <span className="ca-gov-icon-flickr" aria-hidden="true" />
                      <span className="sr-only">Flickr</span>
                    </a>
                  </li>
                  <li>
                    <a href="https://www.pinterest.com/cagovernment/">
                      <span className="ca-gov-icon-pinterest" aria-hidden="true" />
                      <span className="sr-only">Pinterest</span>
                    </a>
                  </li>
                  <li>
                    <a href="https://twitter.com/cagovernment">
                      <span className="ca-gov-icon-twitter" aria-hidden="true" />
                      <span className="sr-only">Twitter</span>
                    </a>
                  </li>
                  <li>
                    <a href="https://www.youtube.com/user/californiagovernment">
                      <span className="ca-gov-icon-youtube" aria-hidden="true" />
                      <span className="sr-only">YouTube</span>
                    </a>
                  </li>
                </ul>
              </div>
            </div>
          </div>

          <div className="copyright">
            <div className="container">
              {this.getCopyRight()}
            </div>
          </div>
        </footer>

        <div className="decoration-last">&nbsp;</div>
      </React.Fragment>
    );
  }
}

Footer.propTypes = {
  contactLink: T.string.isRequired,
};

export const mapStateToProps = createStructuredSelector({
  contactLink: routerSelectors.getContactLink(),
});

const withRedux = connect(mapStateToProps);

export default withRedux(Footer);
