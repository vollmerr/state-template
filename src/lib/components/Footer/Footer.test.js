import React from 'react';
import { shallow } from 'enzyme';

import { Footer as fromLib } from '../..';
import Footer from './Footer';

const props = {
  contactLink: {
    href: 'http://www.google.com',
    text: 'link text',
  },
};

let wrapper;
describe('Footer', () => {
  beforeEach(() => {
    wrapper = shallow(<Footer {...props} />);
  });

  it('should be exported as part of the library', () => {
    expect(fromLib).toEqual(Footer);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
