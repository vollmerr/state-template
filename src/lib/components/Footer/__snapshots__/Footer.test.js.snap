// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render correctly 1`] = `
<Fragment>
  <footer
    className="global-footer"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="three-quarters"
        >
          <ul
            className="footer-links"
          >
            <li
              key="http://www.ca.gov/Use"
            >
              <A
                href="http://www.ca.gov/Use"
                iconProps={null}
                text="Conditions of Use"
                to=""
              />
            </li>
            <li
              key="http://www.ca.gov/Privacy"
            >
              <A
                href="http://www.ca.gov/Privacy"
                iconProps={null}
                text="Privacy Policy"
                to=""
              />
            </li>
            <li
              key="http://www.ca.gov/Accessibility"
            >
              <A
                href="http://www.ca.gov/Accessibility"
                iconProps={null}
                text="Accessibility"
                to=""
              />
            </li>
            <li>
              <A
                href="http://www.google.com"
                iconProps={null}
                text="link text"
                to=""
              />
            </li>
          </ul>
        </div>
        <div
          className="quarter text-right"
        >
          <ul
            className="socialsharer-container"
          >
            <li
              key="https://www.flickr.com/groups/californiagovernment"
            >
              <A
                href="https://www.flickr.com/groups/californiagovernment"
                iconProps={
                  Object {
                    "name": "flickr",
                    "srOnly": "Flickr",
                  }
                }
                text=""
                to=""
              />
            </li>
            <li
              key="https://www.pinterest.com/cagovernment"
            >
              <A
                href="https://www.pinterest.com/cagovernment"
                iconProps={
                  Object {
                    "name": "pinterest",
                    "srOnly": "Pinterest",
                  }
                }
                text=""
                to=""
              />
            </li>
            <li
              key="https://twitter.com/cagovernment"
            >
              <A
                href="https://twitter.com/cagovernment"
                iconProps={
                  Object {
                    "name": "twitter",
                    "srOnly": "Twitter",
                  }
                }
                text=""
                to=""
              />
            </li>
            <li
              key="https://www.youtube.com/user/californiagovernment"
            >
              <A
                href="https://www.youtube.com/user/californiagovernment"
                iconProps={
                  Object {
                    "name": "youtube",
                    "srOnly": "YouTube",
                  }
                }
                text=""
                to=""
              />
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      className="copyright"
    >
      <div
        className="container d-flex justify-content-between"
      >
        <span>
          Copyright © 2000
        </span>
        <span>
          v0.0.1
        </span>
      </div>
    </div>
  </footer>
  <div
    className="decoration-last"
  >
     
  </div>
</Fragment>
`;
